components/mediation-registry/org.wso2.carbon.mediation.registry/src/main/java/org/wso2/carbon/mediation/registry/WSO2Registry.java

```diff
@@ -625,7 +625,9 @@ public Properties getResourceProperties(String entryKey) {
                     Object value = resourceProperties.get(key);
                     if (value instanceof List) {
                         if (((List) value).size() > 0) {
-                            properties.put(key, ((List) value).get(0));
+                            if (((List) value).get(0) != null) {
+                                properties.put(key, ((List) value).get(0));
```

Comment:
shall we assign this to a variable and re use ((List) value).get(0) ?

Reply:
fixed with https://github.com/wso2/carbon-mediation/pull/1569/commits/09e7221ba92b37e21ea01884a488ac4499c6080c